pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3105,8 +3105,78 @@ func TestReconcileOutOfSyncPipelineRun(t *testing.T) {
 	}
 }
 
+func TestUpdatePipelineRunStatusFromInformer(t *testing.T) {
+	names.TestingSeed()
+
+	pr := tb.PipelineRun("test-pipeline-run",
+		tb.PipelineRunNamespace("foo"),
+		tb.PipelineRunLabel("mylabel", "myvalue"),
+		tb.PipelineRunSpec("", tb.PipelineRunPipelineSpec(
+			tb.PipelineTask("unit-test-task-spec", "", tb.PipelineTaskSpec(v1beta1.TaskSpec{
+				Steps: []v1beta1.Step{{Container: corev1.Container{
+					Name:  "mystep",
+					Image: "myimage"}}},
+			})),
+		)),
+	)
+
+	d := test.Data{
+		PipelineRuns: []*v1beta1.PipelineRun{pr},
+	}
+	prt := NewPipelineRunTest(d, t)
+	defer prt.Cancel()
+
+	wantEvents := []string{
+		"Normal Started",
+		"Normal Running Tasks Completed: 0",
+	}
+
+	// Reconcile the PipelineRun.  This creates a Taskrun.
+	reconciledRun, clients := prt.reconcileRun("foo", "test-pipeline-run", wantEvents, false)
+
+	// Save the name of the TaskRun that was created.
+	taskRunName := ""
+	for k := range reconciledRun.Status.TaskRuns {
+		if taskRunName != "" {
+			t.Fatalf("Expected 1 TaskRun but got more")
+		}
+		taskRunName = k
+	}
+	if taskRunName == "" {
+		t.Fatalf("Expected 1 TaskRun but got none")
+	}
+
+	// Add a label to the PipelineRun.  This tests a scenario in issue 3126 which could prevent the reconciler
+	// from finding TaskRuns that are missing from the status.
+	reconciledRun.ObjectMeta.Labels["bah"] = "humbug"
+	reconciledRun, err := clients.Pipeline.TektonV1beta1().PipelineRuns("foo").Update(prt.TestAssets.Ctx, reconciledRun, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("unexpected error when updating status: %v", err)
+	}
+
+	// The label update triggers another reconcile.  Depending on timing, the PipelineRun passed to the reconcile may or may not
+	// have the updated status with the name of the created TaskRun.  Clear the status because we want to test the case where the
+	// status does not have the TaskRun.
+	reconciledRun.Status = v1beta1.PipelineRunStatus{}
+	if _, err := clients.Pipeline.TektonV1beta1().PipelineRuns("foo").UpdateStatus(prt.TestAssets.Ctx, reconciledRun, metav1.UpdateOptions{}); err != nil {
+		t.Fatalf("unexpected error when updating status: %v", err)
+	}
+
+	reconciledRun, _ = prt.reconcileRun("foo", "test-pipeline-run", wantEvents, false)
+
+	// Verify that the reconciler found the existing TaskRun instead of creating a new one.
+	for k := range reconciledRun.Status.TaskRuns {
+		if k != taskRunName {
+			t.Fatalf("Status has unexpected taskrun %s", k)
+		}
+	}
+}
+
 func TestUpdatePipelineRunStatusFromTaskRuns(t *testing.T) {
 
+	prUID := types.UID("9c04a4d9-33b8-4641-ad11-2aceed91af7b")
```

Comment:
Do these UIDs have to be actual UUIDs to pass some check? Why not something easier to read like `first` and `second`? It's a bit hard to tell these are two different strings at a 
quick glance.

Reply:
I changed the strings to make them more obviously different.